Assets/Vimeo/Scripts/Recorder/VimeoPublisher.cs

```diff
@@ -43,51 +44,48 @@ public string GetVimeoPermalink()
             if (recorder.videoPermalink != null) {
                 if (recorder.defaultShareLink == LinkType.ReviewPage) {
                     return recorder.videoReviewPermalink;
-                } 
-                else {
+                } else {
                     return recorder.videoPermalink;
                 }
-            } 
-             
+            }
+
             return "https://vimeo.com/" + video.id;
         }
 
         public void PublishVideo(string filename)
         {
-            Debug.Log("[VimeoRecorder] Uploading to Vimeo");            
-            vimeoApi.UploadVideoFile(filename);
+            Debug.Log("[VimeoRecorder] Uploading to Vimeo");
+            vimeoUploader.Upload(filename);
         }
-
-
-        private void UploadProgress(string status, float progress)
+        void UploadProgress(string status, float progress)
         {
             if (OnUploadProgress != null) {
                 OnUploadProgress(status, progress);
             }
         }
 
-        private void UploadComplete(string video_uri)
+        void UploadComplete(string video_uri, float number)
         {
             UploadProgress("SavingInfo", .999f);
 
             video = new VimeoVideo("", video_uri);
 
 #if UNITY_2018_1_OR_NEWER
             if (recorder.defaultVideoInput == VideoInputType.Camera360) {
-                vimeoApi.SetVideoSpatialMode("equirectangular", recorder.defaultRenderMode360 == RenderMode360.Stereo ? "top-bottom" : "mono");
+                vimeoUploader.vimeoApi.SetVideoSpatialMode("equirectangular", recorder.defaultRenderMode360 == RenderMode360.Stereo ? "top-bottom" : "mono");
```

Comment:
In short term, let's somehow pass back a reference to the api

Reply:
Instead of passing it back - `VimeoUploader` now extends the `VimeoApi` class - https://github.com/vimeo/vimeo-unity-sdk/pull/46/commits/2427d90deb58cbaf846728cff0a46aec75b7e04b